src/random-seed/random-seed.c

```diff
@@ -242,6 +245,16 @@ static int run(int argc, char *argv[]) {
                         if (r < 0)
                                 log_error_errno(r, "Failed to write seed to /dev/urandom: %m");
                 }
+                /* If we're going to later write out a seed file, initialize a hash state with
+                 * the contents of the seed file we just read, so that the new one can't regress
+                 * in entropy. */
+                if (write_seed_file) {
+                        sha256_init_ctx(&hash_state);
+                        if (k < 0)
+                                k = 0;
+                        sha256_process_bytes(&k, sizeof(k), &hash_state);
+                        sha256_process_bytes(buf, k, &hash_state);
+                }
         }
```

Comment:
what's the rationale of including the size of the seed in the hash?

I presume there's a good reason for that. Can you add a comment about it to the line?

Reply:
The idea is just to distinguish "bake race" from "baker ace" type of thing. Probably not vitally important here but seems hygienic to do.